src/tainting/Taint_shape.ml

```diff
@@ -0,0 +1,302 @@
+(* Iago Abal
+ *
+ * Copyright (C) 2022-2024 r2c
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+
+open Common
+module T = Taint
+module Taints = T.Taint_set
+
+(*****************************************************************************)
+(* Prelude *)
+(*****************************************************************************)
+
+let base_tag_strings = [ __MODULE__; "taint" ]
+let _tags = Logs_.create_tags base_tag_strings
+let warning = Logs_.create_tags (base_tag_strings @ [ "warning" ])
+let error = Logs_.create_tags (base_tag_strings @ [ "error" ])
+
+(*****************************************************************************)
+(* Types *)
+(*****************************************************************************)
+
+module Fields = Map.Make (struct
+  type t = T.offset
+
+  let compare = T.compare_offset
+end)
+
+type shape =
```

Comment:
This requires far more explanation. What is a shape, maybe a reference to wikipedia
or some chapter of some famous PA book.